src/workerd/tests/bench-builtin-modules.c++

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include <workerd/tests/bench-tools.h>
+#include <workerd/tests/test-fixture.h>
+
+// A benchmark for loading built-in modules.
+
+namespace workerd {
+
+static CompatibilityFlags::Builder createCompatibilityFlags() {
+  static capnp::MallocMessageBuilder flagsBuilder;
+  static auto featureFlags = flagsBuilder.initRoot<CompatibilityFlags>();
+  featureFlags.setNodeJsCompat(true);
+  featureFlags.setNodeJsCompatV2(true);
+  return featureFlags;
+}
+
+class BuiltinModulesBenchmark: public benchmark::Fixture {
+public:
+  virtual ~BuiltinModulesBenchmark() noexcept(true) {}
+
+  void SetUp(benchmark::State& state) noexcept(true) override {
+    static auto flags = createCompatibilityFlags();
+    TestFixture::SetupParams params = {.featureFlags = flags.asReader(), .mainModuleSource = R"(
+        import assert from "node:assert";
+        export default {
+          async fetch(request) {
+            let buffer;
+            for (let i = 0; i < 100; i++) {
+              buffer = await import("node:buffer");
+            }
```

Comment:
Note that the `import(...)` will only be evaluated once, then cached, which means this is really only benchmarking the initial load once then the other 99 times we're just pulling the result from v8's cache.

Reply:
So, higher the count, higher the performance impact would be for compile cache. What about having this call only 2 times?

Reply:
I think what you actually need here is to have the test deploy two workers. One is a primer... you send a single request to it to perform the initial compile of `node:buffer` that generates the compile cache. Then, you send a single request to the second worker that also imports `node:buffer`. That one should see a reduced completion time on the very first `await import`.